agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/TelemetryClient.java

```diff
@@ -155,27 +158,26 @@ public void trackAsync(TelemetryItem telemetry) {
     MonitorDomain data = telemetry.getData().getBaseData();
     if (data instanceof MetricsData) {
       MetricsData metricsData = (MetricsData) data;
-      List<MetricDataPoint> filteredPoints =
-          metricsData.getMetrics().stream()
-              .filter(
-                  point -> {
-                    String metricName = point.getName();
-                    if (nonFilterableMetricNames.contains(metricName)) {
-                      return true;
-                    }
-                    for (MetricFilter metricFilter : metricFilters) {
-                      if (!metricFilter.matches(metricName)) {
-                        return false;
-                      }
-                    }
-                    return true;
-                  })
-              .collect(Collectors.toList());
-
-      if (filteredPoints.isEmpty()) {
+      if (metricsData.getMetrics().isEmpty()) {
+        // this is unexpected
+        logger.debug("MetricsData has no metric point");
+        return;
+      }
+      MetricDataPoint point = metricsData.getMetrics().get(0);
+      String metricName = point.getName();
+      if (!nonFilterableMetricNames.contains(metricName)) {
+        for (MetricFilter metricFilter : metricFilters) {
+          if (!metricFilter.matches(metricName)) {
+            // user configuration filtered out this metric name
+            return;
+          }
+        }
+      }
+
+      if (!Double.isFinite(point.getValue())) {
+        // breeze doesn't like these values
```

Comment:
do we drop that data point and then send the rest to breeze? here because of bad NaN we drop the whole metric. i wonder how other sdks handle it.

Reply:
unless that data point is the key dimension for that metric. it makes sense to drop it. without it metric doesn't make sense.

Reply:
ya, metric value is required

Reply:
> I wonder how other sdks handle it

https://github.com/microsoft/Telemetry-Collection-Spec/issues/29